libkbfs/bsplitter_simple.go

```diff
@@ -147,3 +150,36 @@ func (b *BlockSplitterSimple) ShouldEmbedBlockChanges(
 	bc *BlockChanges) bool {
 	return bc.SizeEstimate() <= b.blockChangeEmbedMaxSize
 }
+
+// SplitDirIfNeeded implements the BlockSplitter interface for
+// BlockSplitterSimple.
+func (b *BlockSplitterSimple) SplitDirIfNeeded(block *DirBlock) (
+	[]*DirBlock, *StringOffset) {
+	if block.IsIndirect() {
+		panic("SplitDirIfNeeded must be given only a direct block")
+	}
+
+	if b.maxDirEntriesPerBlock == 0 ||
+		len(block.Children) <= b.maxDirEntriesPerBlock {
+		return []*DirBlock{block}, nil
+	}
+
+	// Sort the entries and split them down the middle.
+	names := make([]string, 0, len(block.Children))
+	for name := range block.Children {
+		names = append(names, name)
+	}
+
+	sort.Strings(names)
+	// Delete the second half of the names from the original block,
+	// and add to the new block.
+	newBlock := NewDirBlock().(*DirBlock)
```

Comment:
I'm thinking that `NewDirBlock()` should just return a `*DirBlock` instead of a `Block`; or do you depend on the nil interface behavior somewhere?

Reply:
Not sure what you mean by "nil interface", but `getBlockHelperLocked()` takes a function that returns a new `Block`, and `NewDirBlock` is one thing that's passed to it.  (It's worked that way forever.)

Reply:
nil interface = `(Block)(nil)`, which is different from `(*DirBlock)(nil)`.

i.e. `((Block)(nil) == (*DirBlock)(nil)) == false`

Reply:
https://play.golang.org/p/EZ9-nxvjjGN

Reply:
Oh i think I see.  I depend on the function type of `NewDirBlock` to return exactly a `Block`, so that I can pass it into `getBlockHelperLock`.  To extend your example, this wouldn't work if I changed it to return a `Block`: https://play.golang.org/p/vhmzqPo1z34

Reply:
Understood. That answers my question, thanks!